### PR TITLE
ENH: add np.polynomial.chebyshev.chebinterp()

### PR DESCRIPTION
**Pull request by ghost**
...to calculate the polynomial interpolation of a function
using a series of Chebyshev polynomials. It returns the coefficients
of that series starting from the 0th order coefficient, like the 
other methods. This method is useful e.g. for estimating roots before 
running a root-finding method.

### CODE REVIEW

File: numpy/polynomial/chebyshev.py [hunk 1]
```diff
@@ -1830,6 +1831,74 @@ def chebcompanion(c):
...
+    >>> import numpy as np
+    >>> import numpy.polynomial.chebyshev as C
+    >>> def f(x):
+    ...   return np.tanh(x)+0.5
+    ... 
+    >>> C.chebinterp(f, [0, 2*np.pi], 8)
+    array([ 1.30198814,  0.3540514 , -0.25209078,  0.14032589, -0.05706644,
+            0.01226391,  0.00367748, -0.00455335])
+
+    .. versionadded::1.13.0
```
Comment (charris) [inline numpy/polynomial/chebyshev.py]: Too late for that, 1.14.0 is the earliest.

**Comment by uoaei** [comment]
@charris I could easily put a little wrapper in the Chebyshev class to call this method with the domain interval already fed into the params.

I'll make those changes and submit another pull request against the master branch.

**Comment by charris** [comment]
Oops ;) Something went wrong. First, you don't want to work in `maintenance/1.13.x`, and second, don't merge master into your branch, which I suspect you did. Probably the easiest way to fix this is to checkout a work branch, `git checkout -b chebyshev-interpolation master` and then `git cherry-pick <previous-work-hash>`, which should get you a branch with your work that you can use for a new PR. Cleaning up your `maintenance/1.13.x` branch requires `git reset --hard <last valid commit hash>`. Your new PR also has problems and isn't in a work branch, which makes the fixing up more of a problem. Probably a `git reset <last valid hash>` will leave your changed files hanging about and you can then check out a new branch and commit them after fixing up the merge/rebase artifacts.

**Comment by eric-wieser** [comment]
Closing in favor of #9198

**Pull request closed without merging**